### PR TITLE
docs(firestore): incorporate a var to sample, improve docstrings

### PR DESCRIPTION
**Pull request by crwilcox**
We received feedback that it was hard to understand what cursorStr was. This pulls the variable into the sample along with a doc string. Also adds a docstring to nextCursor

### CODE REVIEW

File: datastore/snippets/snippet_test.go [hunk 1]
```diff
@@ -507,8 +507,10 @@ func SnippetIterator_Cursor() {
 	ctx := context.Background()
 	client, _ := datastore.NewClient(ctx, "my-proj")
 	defer client.Close()
-	cursorStr := ""
 	// [START datastore_cursor_paging]
+	// cursorString is a cursort to start querying at.
```
Comment (hongalex) [inline datastore/snippets/snippet_test.go:511]: nit: s/cursort/cursor ?

**Review by hongalex — APPROVED** [review]

**Review by tbpg — APPROVED** [review]
This is an improvement, but I still find this sample extremely confusing.

**Review by codyoss — APPROVED** [review]

**Review by codyoss — APPROVED** [review]

**Review by craiglabenz — APPROVED** [review]
LGTM

**Comment by crwilcox** [comment]
@tbpg I agree. The only thought I have is to name `cursorStr` to `previousCursorStr`. If you think that would be better I am glad to change that as well.

**Comment by tbpg** [comment]
I think the name is fine. The sample might just need a rewrite overall following new guidelines.

**Pull request merged**